src/engine/plugins/tile.cpp

```diff
@@ -439,6 +445,14 @@ void encodeVectorTile(const datafacade::ContiguousInternalMemoryDataFacadeBase &
                 use_line_value(forward_weight);
                 use_line_value(reverse_weight);
 
+                std::uint32_t forward_rate =
+                    static_cast<std::uint32_t>(round(length / forward_weight * 10.));
+                std::uint32_t reverse_rate =
+                    static_cast<std::uint32_t>(round(length / reverse_weight * 10.));
+
+                use_line_value(forward_rate);
```

Comment:
Yeah - Mapbox GL does some ugly formatting on floats https://github.com/mapbox/mapbox-gl-js/pull/3358, so some of the reasoning for using ints vs doubles in this tile encoder is driven by rendering ugliness :-(